server/src/internalClusterTest/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java

```diff
@@ -1248,6 +1248,33 @@ public void testAbortNotStartedSnapshotWithoutIO() throws Exception {
         assertThat(getRepositoryData(repoName).getGenId(), is(0L));
     }
 
+    public void testStartWithSuccessfulShardSnapshotPendingFinalization() throws Exception {
```

Comment:
Sorry for making this so complicated, but this required 4 new tests I'm afraid because you can have all possible combinations of clone or snapshot ready for a shard but not finalized and then wanting to run either a new clone or new snapshot => 2 x 2 = 4 possible buggy scenarios.